fpdf/fpdf.py

```diff
@@ -4004,18 +3934,108 @@ def preload_image(self, name, dims=None):
             self.images[name] = info
         return name, img, info
 
+    @staticmethod
+    def _limit_to_aspect_ratio(x, y, w, h, asp_w, asp_h):
+        """
+        Make an image fit within a bounding box, maintaining its proportions.
+        In the reduced dimension it will be centered within tha available space.
+        """
+        ratio = asp_w / asp_h
+        if h * ratio < w:
+            nw = h * ratio
```

Comment:
Could we use clearer variable names, please?

Why the `asp_` prefix?

`nw` / `nh` could maybe be renamed `new_w` / `new_x`?

---

Some good reads on variable naming:

https://a-nickels-worth.blogspot.com/2016/04/a-guide-to-naming-variables.html
> * As coders our job is to communicate to human readers, not computers.
> * Names should communicate the coder's intent so the reader doesn't have to try to figure it out.
> * Boilerplate must be minimized, because it drains reviewers' ability to concentrate on the code.
> * Don't Put the Type in the Name
> * Use Teutonic Names Most of The Time
> * Move Simple Comments Into Variable Names: We prefer good names over comments but can't replace all comments.
> * Avoid Over-used Cliches: val, value, result, res, retval, tmp, temp, count, str
> * Use Idioms Where Meaning is Obvious
> * May Use Short Names Over Short Distances When Obvious -> I DISAGREE
> * Remove Thoughtless One-Time Variables -> I DISAGREE
> * Use Short OTVs to Break Up Long Lines & Complicated Expressions

From Clean Code by Roger Martin

> Take the time to choose descriptive names. Naming things is a great power, it comes with great responsabilities [CC-N1]
> - Some good characteristics : intention-revealing, pronounceable, searchable. [CC-Chapt1]
> - Don't use Hungarian notation or member prefixes [CC-Chapt1]
> - Use noun/noun phrase naming for types, verb/verb phrases for methods, not only adjectives. Avoid words like Manager, Processor, Data or Info. [CC-Chapt1]
> - Don't pick names that communicate implementationi [CC-N2]
> - Use standard nomenclatures where possible [CC-N3]
> - Unambigous names [CC-N4]
> - Use long names for long scopes : 'i' is ok for small loops [CC-N5]
> - Names should describe side-effects [CC-N7]
> - Function names should say what they do : you can tell it from a call, without looking at the definition [CC-G20] Common smells:
>     * Obvious behaviour is unimplemented [CC-G2]
>     * Incorrect behaviour at the boundaries [CC-G3]

Reply:
Context matters.
We're dealing with either changing or maintaining the aspect ratio of the image here. In this context, it seems rather obvious to me that `asp_w` and `asp_h` are the dimensions that define the aspect ratio.
This is a religious topic for many, but most guidelines agree that variables with very short scope can also have very short names.
Of course I have no problem with spelling the names out longer and make them really screamingly obvious... 😱

Reply:
> Of course I have no problem with spelling the names out longer and make them really screamingly obvious... 😱 

Quoting the [Zen of Python](https://peps.python.org/pep-0020/): "Explicit is better than implicit"

>  We're dealing with either changing or maintaining the aspect ratio of the image here. In this context, it seems rather obvious to me that `asp_w` and `asp_h` are the dimensions that define the aspect ratio.

I'm not a native English speaker, and I can honestly say that I did not understand the naming yesterday...
But maybe I was a bit tired 🙂